packages/library-base/src/commonMain/kotlin/io/realm/internal/util/Channel.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2021 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.internal.util
+
+import kotlinx.coroutines.channels.Channel
+
+// Conventional try-with-resource wrapper for channels
+public inline fun <T : Channel<*>, R> T.autoClose(block: (channel: T) -> R): R {
```

Comment:
Nitpick: Would `use` be better and fit how try-with-resources is normally called?

Reply:
I had some issues with autocompletion trying to import `kotlin.io.use` instead and was annoyed by the overlap, but might be that it was because I accidentially trigger that import on from experiments on `File` os something like that. I have renamed it to `use`, then we can see if it causes issues.